cluster/engine.go

```diff
@@ -217,8 +217,13 @@ func (e *Engine) StartMonitorEvents() {
 		if err := <-ec; err != nil {
 			log.WithFields(log.Fields{"name": e.Name, "id": e.ID}).WithError(err).Error("error monitoring events, will restart")
 			if !strings.Contains(err.Error(), "EOF") {
-				// failing node reconnect should use back-off strategy
-				<-e.refreshDelayer.Wait(e.getFailureCount())
+				// failing node reconnect should use back-off strategy to avoid frequent reconnect
+				retryInterval := e.getFailureCount() + 1
```

Comment:
The random delay was to avoid synchronization on engine reconnect. For example, hundreds of nodes reboot due to power interrupt. With the randomization, their reconnect (and subsequent refresh) would be spread out. The event stream reconnect doesn't have this problem. So I don't think randomization has effect here.